WalletWasabi.Daemon/PersistentConfig.cs

```diff
@@ -12,101 +13,160 @@
 	public bool JsonRpcServerEnabled { get; init; }
 
 	[DefaultValue("")]
 	[JsonProperty(PropertyName = "JsonRpcUser", DefaultValueHandling = DefaultValueHandling.Populate)]
+	[System.Text.Json.Serialization.JsonPropertyName("JsonRpcUser")]
 	public string JsonRpcUser { get; init; } = "";
 
 	[DefaultValue("")]
 	[JsonProperty(PropertyName = "JsonRpcPassword", DefaultValueHandling = DefaultValueHandling.Populate)]
+	[System.Text.Json.Serialization.JsonPropertyName("JsonRpcPassword")]
 	public string JsonRpcPassword { get; init; } = "";
 
 	[JsonProperty(PropertyName = "JsonRpcServerPrefixes")]
+	[System.Text.Json.Serialization.JsonPropertyName("JsonRpcServerPrefixes")]
 	public string[] JsonRpcServerPrefixes { get; init; } = new[]
 	{
 		"http://127.0.0.1:37128/",
 		"http://localhost:37128/"
 	};
 
 	[JsonProperty(PropertyName = "DustThreshold")]
+	[System.Text.Json.Serialization.JsonPropertyName("DustThreshold")]
 	[JsonConverter(typeof(MoneyBtcJsonConverter))]
+	[System.Text.Json.Serialization.JsonConverter(typeof(MoneyBtcJsonConverterNg))]
 	public Money DustThreshold { get; init; } = DefaultDustThreshold;
 
 	[JsonProperty(PropertyName = "EnableGpu")]
+	[System.Text.Json.Serialization.JsonPropertyName("EnableGpu")]
 	public bool EnableGpu { get; init; } = true;
 
 	[DefaultValue("CoinJoinCoordinatorIdentifier")]
 	[JsonProperty(PropertyName = "CoordinatorIdentifier", DefaultValueHandling = DefaultValueHandling.Populate)]
+	[System.Text.Json.Serialization.JsonPropertyName("CoordinatorIdentifier")]
 	public string CoordinatorIdentifier { get; init; } = "CoinJoinCoordinatorIdentifier";
 
+	public bool DeepEquals(PersistentConfig other)
+	{
+		return
+			Network == other.Network &&
+			MainNetBackendUri == other.MainNetBackendUri &&
+			TestNetBackendUri == other.TestNetBackendUri &&
+			RegTestBackendUri == other.RegTestBackendUri &&
+			MainNetCoordinatorUri == other.MainNetCoordinatorUri &&
+			TestNetCoordinatorUri == other.TestNetCoordinatorUri &&
+			RegTestCoordinatorUri == other.RegTestCoordinatorUri &&
+			UseTor == other.UseTor &&
+			TerminateTorOnExit == other.TerminateTorOnExit &&
+			DownloadNewVersion == other.DownloadNewVersion &&
+			StartLocalBitcoinCoreOnStartup == other.StartLocalBitcoinCoreOnStartup &&
+			StopLocalBitcoinCoreOnShutdown == other.StopLocalBitcoinCoreOnShutdown &&
+			LocalBitcoinCoreDataDir == other.LocalBitcoinCoreDataDir &&
+			MainNetBitcoinP2pEndPoint.Equals(other.MainNetBitcoinP2pEndPoint) &&
+			TestNetBitcoinP2pEndPoint.Equals(other.TestNetBitcoinP2pEndPoint) &&
+			RegTestBitcoinP2pEndPoint.Equals(other.RegTestBitcoinP2pEndPoint) &&
+			JsonRpcServerEnabled == other.JsonRpcServerEnabled &&
+			JsonRpcUser == other.JsonRpcUser &&
+			JsonRpcPassword == other.JsonRpcPassword &&
+			JsonRpcServerPrefixes.SequenceEqual(other.JsonRpcServerPrefixes) &&
+			DustThreshold == other.DustThreshold &&
+			EnableGpu == other.EnableGpu &&
+			CoordinatorIdentifier == other.CoordinatorIdentifier;
+	}
```

Comment:
Why not use `JsonSerilaizer.Serialize` on both `other` and `this`, and compare the strings here?
That way we don't have to add optional future properties here.

Reply:
Understand. Knowing that we rarely do add here new things, it's okay like this for the time being.

Reply:
I understand, I meant "for the time being" as until we don't add something new here.

Reply:
I think the `JsonRpcServerPrefixes` was a mistake. You can remove it, that would simplify many things not only in this pr.